### ISSUE TITLE
Issue when adding song to play next

### ISSUE THREAD

**Issue by sn0am** [blackcandy-org/blackcandy#387]
When a song is already in the queue, doing "Play Next" on the song through the library gives an error stating "Add to playlist failed. Song is already within the playlist".

In such a case, the song needs to be removed from the queue entirely and then "Play Next" needs to be clicked so it can be added to the queue to be played next. 

It would make more sense to have the song moved within the queue to the next song when "Play Next" is pressed from within the library, rather than attempting to add the song again to the library. Or maybe existing instance can be deleted from the queue and new instance could be added as the next song in the queue.

**Comment by aidewoode** [comment]
You are right, it's more sense than just show an error message.  And for the 'Play Last', it should also have the same behavior. But right now you can also move any songs in the playing queue to reorder.
